addons/Lookup/Lookup.lua

```diff
@@ -0,0 +1,190 @@
+				player = 'https://www.ffxiah.com/search/player?name=%s';
+			};
+			google = 'https://www.google.com/search?q=%s';
+		};
+	})
+end
+
+function initialize_ids()
+	ids = {
+		items = {};
+		zones = {};
+	}
+	for item in res.items:it() do
+	    ids.items[item.name] = item.id
+	    ids.items[item.name_log] = item.id
+	end
+	for zone in res.zones:it() do
+			ids.zones[zone.name] = zone.id
+	end
+end
+
+function find_id(name)
+	return {
+		item = ids.items[name];
+		zone = ids.zones[name];
+	}
+end
+
+function get_name(id, list)
+	for key, value in next, list do
+		if value == id then
+			return key
+		end
+	end
+	return nil
+end
+
+function translate(str)
+	return windower.convert_auto_trans(str)
+end
+
+function get_selection(str)
+	local target = str:match('<(.+)>')
+	if target == 'job' then
+		str = windower.ffxi.get_player().main_job_full
+	elseif target == 'subjob' or target == 'sj' then
+		str = windower.ffxi.get_player().sub_job_full
+	elseif target == 'zone' or target == 'area' then
+		if windower.ffxi.get_info().mog_house then
+			str = 'Mog House'
+		else
+			str = get_name(windower.ffxi.get_info().zone, ids.zones)
+		end
+	elseif target == 'lastitem' or target == 'item' then
+		str = get_name(last_item, ids.items)
+	elseif target ~= nil then
+		local mob = windower.ffxi.get_mob_by_target(str:match('<(.+)>'))
+		if mob ~= nil then
+			str = mob.name
+		else
+			str = nil
+		end
+	end
+	return str
+end
+
+function set_default(term)
+	settings.default = term
+	config.save(settings, 'all')
+end
+
+function set_last_item(bag, index, id, count)
+	if bag == 0 then
+		last_item = id
+	end
+end
+
+function process_command(command, term)
+	if term == nil or term == '' then
+		term = command
+		command = settings.default
+	else
+		command = command:lower()
+	end
+
+	if command == 'default' then
+		set_default(term)
+		return
+	end
+
+	term = translate(term)
+	term = get_selection(term)
+
+	if term == nil or term == '' then
+		return
+	end
+
+	local url
+	local id = find_id(term)
+	if command == 'ffxiclopedia' or command == 'ffxi' or command == 'wikia' then
+		url = settings.url.ffxiclopedia:format(term)
+	elseif command == 'bg' or command == 'bgwiki' or command == 'bg-wiki' then
+		url = settings.url.bgwiki:format(term)
+	elseif command == 'db' or command == 'ffxidb' then
+		if id.item ~= nil then
+			url = settings.url.ffxidb.item:format(id.item)
+		elseif id.zone ~= nil then
+			url = settings.url.ffxidb.zone:format(id.zone)
+		else
+			url = settings.url.ffxidb.search:format(term)
+		end
+	elseif command == 'ah' or command == 'ffxiah' then
+		if id.item ~= nil then
+			url = settings.url.ffxiah.item:format(id.item)
+		else
+			url = settings.url.ffxiah.search:format(term)
+		end
+	elseif command == 'ahp' or command == 'ffxiahp' or command == 'ffxiahplayer' then
+		url = settings.url.ffxiah.player:format(term)
+	elseif command == 'google' then
+		url = settings.url.google:format(term)
+	end
```

Comment:
You do a hard-coded URL lookup. I would design it in such a way that lookup goes automatically by what is specified in the settings file. That way users can add their own sites as well. The way it is now I don't see why it needs to be in user-facing settings at all, since the structure is hard-coded, and it makes little sense to adjust the URLs.

This would remove some more ambiguity as for what to specify as a command, the same thing I mentioned above.